src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/KeyConverters/StringKeyConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class StringKeyConverter : KeyConverter<string>
+    {
+        public override string Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            throw new NotImplementedException();
```

Comment:
I believe we should be throwing `InvalidOperationException` here instead. `NotImplemented` is more for "will probably implement at a future time". `NotSupported` is a thought, but I believe `InvalidOperation` is the most correct here.